src/paths/distances.c

```diff
@@ -212,3 +238,70 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
 
     return 0;
 }
+
+/**
+ * \function igraph_pseudo_diameter
+ * \brief Approximation and lower bound of diameter.
+ *
+ * This algorithm finds a pseudo-peripheral vertex and returns its
+ * eccentricity. This value can be used as an approximation
+ * and lower bound of the diameter of a graph.
+ *
```

Comment:
It should be noted that this function computes the pseudo-diameter per-component. This is another reason to allow choosing the starting vertex: it will use the component to which that vertex belongs.

It should not be difficult to detect if the graph is disconnected during the eccentricity calculation, but let's think about whether it is useful to detect that case at all.

Reply:
My best guess is yes, people don't always know what kind of graph they're working with, and not realizing the graph is disconnected can lead you to the wrong interpretation of the results. Disconnectedness seems like something that is always relevant when running this function, and the main reason not to check is if you already know up front if the graph is or isn't disconnected. I'm not really a user though :slightly_smiling_face:

Reply:
Similar to `igraph_diameter`, we can add an `unconn` parameter that controls what happens. If true, the calculation will be done for the component of the starting vertex If false, infinity should be returned for a disconnected graph.

Reply:
Another view is that the pseudo-diameter gives a lower bound for the diameter: this returning a finite result is acceptable when the true result is infinity.